pyglet/graphics/shader.py

```diff
@@ -236,31 +236,128 @@ def __repr__(self):
         return f"Attribute(name='{self.name}', location={self.location}, count={self.count})"
 
 
+class _UniformArray:
+    """Wrapper of the GLSL array data inside a Uniform.
+    Allows access to get and set items for a more Pythonic implementation.
+    Types with a length longer than 1 will be returned as tuples as an inner list would not support individual value
+    reassignment. Array data must either be set in full, or by indexing."""
+
+    _slots_ = ('uniform', 'gl_type', 'gl_getter', 'gl_setter', 'is_matrix', 'dsa', '_c_array', '_ptr')
```

Comment:
Looks like the `_slots_` should be `__slots__`, and some of the name are missing the leading underscore.